framework/filters/HostControl.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\filters;
+
+use Yii;
+use yii\base\ActionFilter;
+use yii\web\NotFoundHttpException;
+
+/**
+ * HostControl provides simple control over requested host name.
+ *
+ * This filter provides protection against ['host header' attacks](https://www.acunetix.com/vulnerabilities/web/host-header-attack),
+ * allowing action execution only for specified host names.
+ *
+ * Application configuration example:
+ *
+ * ```php
+ * return [
+ *     'as hostControl' => [
+ *         'class' => 'yii\filters\HostControl',
+ *         'allowedHosts' => [
+ *             'example.com',
+ *             '*.example.com',
+ *         ],
+ *     ],
+ *     // ...
+ * ];
+ * ```
+ *
+ * Controller configuration example:
+ *
+ * ```php
+ * use yii\web\Controller;
+ * use yii\filters\HostControl;
+ *
+ * class SiteController extends Controller
+ * {
+ *     public function behaviors()
+ *     {
+ *         return [
+ *             'hostControl' => [
+ *                 'class' => HostControl::className(),
+ *                 'allowedHosts' => [
+ *                     'example.com',
+ *                     '*.example.com',
+ *                 ],
+ *             ],
+ *         ];
+ *     }
+ *
+ *     // ...
+ * }
+ * ```
+ *
+ * > Note: the best way to restrict allowed host names is usage of the web server 'virtual hosts' configuration.
+ * This filter should be used only if this configuration is not available or compromised.
+ *
+ * @author Paul Klimov <klimov.paul@gmail.com>
+ * @since 2.0.11
+ */
+class HostControl extends ActionFilter
+{
+    /**
+     * @var array|\Closure|null list of host names, which are allowed.
+     * Each host can be specified as a wildcard pattern. For example:
+     *
+     * ```php
+     * [
+     *     'example.com',
+     *     '*.example.com',
+     * ]
+     * ```
+     *
+     * This field can be specified as a PHP callback of following signature:
+     *
+     * ```php
+     * function (\yii\base\Action $action) {
+     *     //return array of strings
+     * }
+     * ```
+     *
+     * where `$action` is the current [[\yii\base\Action|action]] object.
+     *
+     * If this field is not set - no host name check will be performed.
+     */
+    public $allowedHosts;
+    /**
```

Comment:
wouldnt a naming $allowedHostPatterns be more appropiate if you look at the description?